addon/pods/components/rui-validatable-input/component.js

```diff
@@ -8,35 +8,103 @@ const {
-    return this.get('valuePath') in attrsChanged
+  // Validation binding attributes with check for async settings
+  // if async is true, don't display either until `_didValidate`
+
+  isInvalid: computed('_isInvalid', '_didValidate', function() {
+    if (this.get('async')) {
+      return this.get('_isInvalid') && this.get('_didValidate')
+    }
+    return this.get('_isInvalid')
+  }),
+
+  isValid: computed('_isValid', '_didValidate', function() {
+    if (this.get('async')) {
+      return this.get('_isValid') && this.get('_didValidate')
+    }
+    return this.get('_isValid')
   }),
-  didValidate: computed.oneWay('targetObject.didValidate'),
-  hasContent: computed.notEmpty('value'),
-  isInvalid: computed.oneWay('validation.isInvalid'),
-  isValid: computed.and('didChange', 'hasContent', 'validation.isValid', 'notValidating'),
-  notValidating: computed.not('validation.isValidating'),
-  showErrorClass: computed.and('notValidating', 'showMessage', 'validation'),
-  showMessage: computed('validation.isDirty', 'isInvalid', 'didValidate', function() {
-    return (this.get('validation.isDirty') || this.get('didValidate')) && this.get('isInvalid')
-  })
 
+  // This method ensures that if presence is required
+  // But the initial value is undefined or an empty string
+  // (the case with new records) we can still validate even though
+  // the `_didChange` attr would report it had not
+
+  _validatePresenceWithEmptyDefault: computed('value', function() {
+    const validationOptions = Object.keys(
+      this.get(`model.validations.attrs.${this.get('valuePath')}.options`))
+
+    return (validationOptions.indexOf('presence') !== -1) &&
+      (this.get('value') === undefined) ||
+      (this.get('value') === '')
+  }),
+
+  // The attribute has changed UNLESS
+  // the validator includes presence and
+  // the value is undefined or empty
+
+  _didChange: computed(
+    'value',
+    'model.hasDirtyAttributes',
+    function() {
+      if (this.get('_validatePresenceWithEmptyDefault')) { return true }
+
+      const attrsChanged = this.get('model') ?
+        this.get('model').changedAttributes() : {}
+      return this.get('valuePath') in attrsChanged
+    }
+  ),
+
+  // Checks for in processing statuses
+  // Ensures validations don't show when:
+  // input is clean, is validating, model is saving
+
+  _validatable: computed(
+    '_didChange',
+    'validation.isValidating',
+    'model.isSaving',
+    function() {
+      return this.get('_didChange') &&
+      !this.get('validation.isValidating') &&
+      !this.get('model.isSaving')
+    }
+  ),
+
+  // Base values that compute validation state
+
+  _isInvalid: computed.and('validation.isInvalid', '_validatable'),
+  _isValid: computed.and('validation.isValid', '_validatable'),
+
+  // Checks property `_didValidate` on controller
+  // Used to only show validation if `async` is true
+
+  _didValidate: computed.oneWay('targetObject.didValidate')
```

Comment:
There is a good bit of discussion on this topic.  Either way we will likely need to manually set it on the model or the targetObject.  I did a bit of research into hooking to store events but this still won't solve the non-model based validations.  But I agree that it would likely be a little less brittle?

I wish this was set when .validate() is called.